### ISSUE TITLE
Suggest NetDialTimeout be part of Dialer NOT Global

### ISSUE THREAD

**Issue by krolaw** [go-mail/gomail#40]
For systems like Google's app engine (standard), out going sockets need a context (per incoming http connection). see: https://godoc.org/google.golang.org/appengine/socket
This makes it really hard when the netDialout command is global, since simultaneous requests are coming in.  It's much more practical than before though.

**Comment by ivy** [comment]
@krolaw Agreed. I would accept a pull request adding a `NetDialTimeout` field to `Dialer` and using the global `NetDialTimeout` as a fallback.

**Edit:** I see you've already done this in PR #41. I'll review and see if we can get this merged. 😄

**Comment by krolaw** [comment]
Hi, any movement on this? Thanks.
